packages/react/src/components/Authenticator/hooks/useAuthenticator/index.tsx

```diff
@@ -179,12 +150,21 @@ export const useAuthenticator = (selector?: Selector) => {
     return areArrayValuesEqual(prevDepsArray, nextDepsArray);
   };
 
-  const state = useSelector(service, xstateSelector, comparator);
+  const facade = useSelector(
+    service,
+    xstateSelector,
+    comparator,
+    /**
+     * Below parameter was needed to avoid "The result of getSnapshot should be
+     * cached to avoid an infinite loop" error on React 18
+     */
+    (interpreter) => interpreter.getSnapshot()
```

Comment:
I'm not entirely certain what is happening here but what is preventing us from caching the result of `getSnapshot`? Prefer to avoid adding more workarounds (maybe this is not a workaround, but that's what i inferred from the comment)

Reply:
I'm not seeing the same error if I delete this `getSnapshot` parameter from the `useSelector` call. The [xstate docs](https://xstate.js.org/docs/packages/xstate-react/#useselector-actor-selector-compare-getsnapshot) mention the parameter is optional, but don't really give an example of when you would want to specify it. Are you able to walk me through replicating that error you saw?

Reply:
Oops, I think I worded that incorrectly. This is not a workaround, and is an intended use of the fourth argument in [`useSelector`](https://xstate.js.org/docs/packages/xstate-react/#useselector-actor-selector-compare-getsnapshot) here. 

If we don't pass in this fourth argument, xstate guesses the usage and passes in a generic `getSnapshot` defined here https://github.com/statelyai/xstate/blob/c0b787d2a2334665af0ecb4ebc1ce5653fa3725c/packages/xstate-react/src/useSelector.ts#L12-L17. This led to the error mentioned in the comment, so I tightened it up by which `getSnapshot` our interpreter uses.

Reply:
@slaymance, sorry I commented before I saw yours!

I saw this error on our `examples/next` without the fourth option. Let me try to repro again in a clean environment.

Reply:
Oh wait, you might be totally correct @slaymance, I don't see that error anymore. Thanks for run testing my code!

Reply:
Removed [7af3c68](https://github.com/aws-amplify/amplify-ui/pull/1798/commits/7af3c68d948a4483c23192d0b69f16adf5ed251f) here, will resolve this.